src/repositories/genre_repository.go

```diff
@@ -0,0 +1,207 @@
+package repositories
+
+import (
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/.gen/KinoTicketSystem/model"
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/.gen/KinoTicketSystem/table"
+	kts_errors "github.com/ELITE-Kinoticketsystem/Backend-KTS/src/errors"
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/managers"
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/models"
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/utils"
+	"github.com/go-jet/jet/v2/mysql"
+	"github.com/google/uuid"
+)
+
+type GenreRepositoryI interface {
+	// Genre
+	GetGenres() (*[]model.Genres, *models.KTSError)
+	GetGenreByName(name *string) (*model.Genres, *models.KTSError)
+	CreateGenre(name *string) *models.KTSError
+	UpdateGenre(genre *model.Genres) *models.KTSError
+	DeleteGenre(genreId *uuid.UUID) *models.KTSError
+
+	// All Movies with all Genres - Grouped by Genre
+	GetGenresWithMovies() (*[]models.GenreWithMovies, *models.KTSError)
+
+	// One Genre with all Movies
+	GetGenreByNameWithMovies(genreName *string) (*models.GenreWithMovies, *models.KTSError)
+}
+
+type GenreRepository struct {
+	DatabaseManager managers.DatabaseManagerI
+}
+
+// Genre
+func (mr *GenreRepository) GetGenres() (*[]model.Genres, *models.KTSError) {
+	var genres []model.Genres
+
+	// Create the query
+	stmt := mysql.SELECT(
+		table.Genres.AllColumns,
+	).FROM(
+		table.Genres,
+	)
+
+	// Execute the query
+	err := stmt.Query(mr.DatabaseManager.GetDatabaseConnection(), &genres)
+	if err != nil {
+		return nil, kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if len(genres) == 0 {
+		return nil, kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return &genres, nil
+}
+
+func (mr *GenreRepository) GetGenreByName(name *string) (*model.Genres, *models.KTSError) {
```

Comment:
For what could this be needed actually?

Reply:
For search

Reply:
Was für eine search? Search nach genres, haben wir eigtl nicht und selbst wenn wäre die nicht wirklich hilfreich, da du mit EQ vergleicht also nur namen die exact gleich sind

Reply:
Ahh jetzt weiß ich warum ich das noch drin hatte. Wenn jemand einen neuen Film anlegt und ein Genre hinzufügt, muss erst kontrolliert werden, ob es das Genre schon gibt und da hat man nur den String. Wenn es den nicht gibt wird er angelegt (create), wenn es ihn schon gibt, wird das Objekt mit der UUID zurückgegeben, welches dann für das Verknüpfen mit dem Film notwendig ist.

Reply:
Hätte gedacht das wird die unabhängig voneinander anlegen, und dann die die genre ids zu dem movie hinzugefügt werden.

Reply:
Dachte wir machen das in dem PopUp

Reply:
So Sachen müssen wir halt absprechen und von dem was wir besprochen haben, dachte ich, wäre das die Lösung gewesen.

Reply:
Genau in dem Popup, aber wenn das popup geschlossen wird wird die request für create genre geschicht. Mit der id die dann returned wird, kann die direkt hinzugefügt werden

Reply:
Sonst müsstest du ja auch actor noch hinzufügen machst du ja auch nicht

Reply:
Removed it